tests/storage/test_id_generators.py

```diff
@@ -398,6 +399,56 @@ async def _get_next_async2() -> None:
         second_id_gen.advance("first", 8)
         self.assertEqual(second_id_gen.get_positions(), {"first": 8, "second": 9})
 
+    def test_multi_instance_empty_row(self) -> None:
+        """Test that reads and writes from multiple processes are handled
+        correctly, when one of the writers starts without any rows.
+        """
+        # Insert some rows for two out of three of the ID gens.
+        self._insert_rows("first", 3)
+        self._insert_rows("second", 4)
+
+        first_id_gen = self._create_id_generator(
+            "first", writers=["first", "second", "third"]
+        )
+        second_id_gen = self._create_id_generator(
+            "second", writers=["first", "second", "third"]
+        )
+        third_id_gen = self._create_id_generator(
+            "third", writers=["first", "second", "third"]
+        )
+
+        self.assertEqual(
+            first_id_gen.get_positions(), {"first": 3, "second": 7, "third": 7}
+        )
```

Comment:
Where do the sevens come from?!

Reply:
Ah, its a bit non obvious: we insert 3 rows with the "first" instance, then 4 rows with the "second" instance, and so the final stream ID is 7

Reply:
oh I though the 4 was setting the position of the second writer to 4. Maybe leave a comment to that effect? Or maybe I should learn to read :shrug: